modules/storages/spec/features/delete_project_storage_and_file_links_spec.rb

```diff
@@ -0,0 +1,74 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2022 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require_relative '../spec_helper'
+
+# Test if the deletion of a ProjectStorage actually deletes related FileLink
+# objects.
+describe 'Delete ProjectStorage with FileLinks', type: :feature, js: true do
+  let(:admin) { create(:admin) }
+  let(:storage) { create(:storage, name: "Storage 1") }
+  let(:project) do
+    create(:project,
+           name: 'Project 1',
+           identifier: 'demo-project',
+           enabled_module_names: %i[storages work_package_tracking])
+  end
+  let(:work_package) { create(:work_package, project: project) }
+  let(:project_storage) { create(:project_storage, storage: storage, project: project) }
+  let(:file_link) { create(:file_link, storage: storage, container_id: work_package.id) }
```

Comment:
```suggestion
  let(:file_link) { create(:file_link, storage: storage, container: work_package) }
```
Does the same but looks somewhat nicer.